pkg/agent/openflow/feature_templates.go

```diff
@@ -0,0 +1,352 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openflow
+
+import (
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+)
+
+// OVS pipelines are generated by a framework called FlexiblePipeline. There are some abstractions introduced in this
+// framework.
+// +--------------+ +--------------+ +--------------+ +--------------+ +--------------+ +--------------+
+// | FeatureTable | | FeatureTable | | FeatureTable | | FeatureTable | | FeatureTable | | FeatureTable |
+// +--------------+ +--------------+ +--------------+ +--------------+ +--------------+ +--------------+
+//         \               |                /                |                 \               /
+//          \              |               /                 |                  \             /
+//           \             |              /                  |                   \           /
+//      +------------------------------------+        +--------------+       +--------------------+
+//      |               stage                |        |     stage    |       |        stage       |
+//      +------------------------------------+        +--------------+       +--------------------+
+//              \                                          /    \                        /
+//               \                                        /      \                      /
+//                \                                      /        \                    /
+//              +------------------------------------------+    +--------------------------+
+//              |                 feature                  |    |          feature         |
+//              +------------------------------------------+    +--------------------------+
+//                                  |                    \                         /
+//                                  |                     \                       /
+//                                  |                      \                     /
+//                     +------------------------+        +-------------------------+
+//                     |       OVS pipeline     |        |       OVS pipeline      |
+//                     +------------------------+        +-------------------------+
+
+// FeatureTable in FlexiblePipeline is the basic unit to build OVS pipelines. It is the enhanced version of ofTable. A
+// FeatureTable can be used by one or more active features. If a FeatureTable is used by any active features, then the
```

Comment:
If FeatureTable doesn't belong to a single feature, could we remove "feature" from its name? It sounds a bit confusing.

And I'm trying to understand the necessity of "stage". It seems to me that in makes the relationship a bit complicated. Without it, "table" is the basic unit of "pipeline", "feature" is an abstraction of a feature which can use several "tables". When developing a new feature, we reuse existing table or add table, and insert flows in these tables, making sure the table is organized in the desired way by having a slice listing all tables in order, then priority is not needed either.
Why stage is needed?

Reply:
There are two reasons:

- "stage" is a set of tables to implement similar functionality. For example, tables in SwitchingStage are only related with layer 2 forwarding and tables in RoutingStage are only related with layer 3 forwarding.
- The order of all "stage" is fixed. When adding a new table, without "stage", we need to consider all existing tables' priority to put the newly added table to desired position, with "stage", we only need to consider all existing tables' priority with the target "stage"